documentation/AuthorizationService.md

```diff
@@ -1,123 +1,245 @@
+    * scope: String. Must be "add_preferences".
+        - Notes: "add_preferences" is the only scope currently supported by the [GPII Client Credentials Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Client_Credentials_Grant).
+* **return:** String. A privileged prefs creator access token. Return an object that contains the error message and http status code if the scope is wrong, or the client is not allowed to add preferences, or the client's OAuth2 client type is not "oauth2ClientCredentials".
+
+#### grantWebPrefsConsumerAuthCode(userId, clientId, redirectUri, selectedPreferences)
+- **description**: Grant an Authorization Code for the specified user, client, redirect URI and selected preferences. We first check to see if we have an existing authorization for the user, client, and redirect URI. If we do, we issue a new code for that authorization. Otherwise we create a new authorization record and a new code.
+- **parameters:** 
+    - userId: String. A system generated unique number that identifies the user.
+    - clientId: String. A system generated unique number that identifies the client.
+    - redirectUri: String. The client redirection URI that the authorization server directs the user-agent to when a authorization is established.
+    - selectedPreferences: Object. An object specifying the preferences that the user has selected to share, in the privacy ontology. An example:
     ```
     {
         "visual-alternatives.speak-text.rate": true,
         "increase-size.appearance.text-size": true
     }
     ```
-* **return:** An authorization code.
+- **return:** An authorization code.
 
-#### revokeAuthorization(userId, authDecisionId)
-* **description**: Revoke an authorization decision. Before the revoke, the function ensures the authorization decision was made by the user.
+#### revokePrivilegedPrefsCreatorAuthorization(authorizationId)
+* **description**: Revoke an authorization granted to a privileged preferences creator.
 * **parameters:** 
-    * userId: Number. A system generated unique number that identifies the user.
-    * authDecisionId: Number. A system generated unique number that identifies the authorization decision.
+    * authorizationId: String. A string representing the id of the client credential authorization record.
 * **return:** None.
 
-#### setSelectedPreferences(userId, authDecisionId, selectedPreferences)
-* **description**: Update the user selected preferences that is saved within the authorization decision. The function verifies the authorization decision is made by the user.
+#### revokeUserAuthorizedAuthorization(userId, authorizationId)
+* **description**: Revoke an authorization. This API is used to revoke these authorization types: GPII app installation authorizations, onboarded solution authorizations, web preferences consumer authorizations.
```

Comment:
If I'm tracing through the code correctly, I don't think that `revokeUserAuthorizedAuthorization` will revoke GPII App Installations.

https://github.com/cindyli/universal/blob/GPII-2435/gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/dbViews/views.json#L35-L37